library/ecp_curves.c

```diff
@@ -5477,6 +5477,204 @@ static int ecp_mod_p256k1(mbedtls_mpi *N)
 }
 #endif /* MBEDTLS_ECP_DP_SECP256K1_ENABLED */
 
-#endif /* !MBEDTLS_ECP_ALT */
+#if defined(MBEDTLS_TEST_HOOKS)
+
+/** Initialise a modulus with hard-coded const curve data.
+ *
+ * \param[out] N    The address of the modulus structure to populate.
+ *                  Must be initialized.
+ * \param[in] id    The mbedtls_ecp_group_id for which to initialise the modulus.
+ * \param[in] ctype The mbedtls_ecp_curve_type identifier for a coordinate modulus (P)
+ *                  or a scalar modulus (N).
+ *
+ * \return          \c 0 if successful.
+ * \return          #MBEDTLS_ERR_ECP_BAD_INPUT_DATA if the given MPIs do not
+ *                  have the correct number of limbs.
+ *
+ * \note            The caller is responsible for the \p N moduli lifecycle.
```

Comment:
This is kinda implicit and true with all setup function. I think it would be helpful to tell explicitly which function needs to be used to free the modulus.

Also, we usually put notes above the parameter list in doxygen documentation.